### DISCUSSION TITLE
Lesson 9: Getting Event Data

### DISCUSSION THREAD

**Discussion by miladad8** [Cyfrin/foundry-full-course-cu#1610] [Q&A] [answered]
Hi. In lesson 9 when we are using an output of an event for testing, we try to store and emit requestId as an event and use it in our test.
Is it possible to get the requestId using Getters function? does it have to be using event and logging functionality?

 sorry if it's a dummy question :)

**Comment by EngrPips** [comment, marked as answer]
Hello @miladad8 , actually Patrick intentionally wrote a test around an event data so as to demonstrate how we would do it in a scenario where testing that event is properly emitted is paramount. and using a getter to get the data of an emitted data wouldn't be the easiest thing to do if at all it is possible so the way Patrick demonstrated in the course seem to be the best approach if you would ask me

**Comment by miladad8** [comment]
thanks a lot!

**Reply by EngrPips** [reply]
Anytime Friend. keep crushing.
